pkg/pods/pod_test.go

```diff
@@ -216,43 +217,35 @@ func TestLogInfo(t *testing.T) {
 }
 
 func TestWriteManifestWillReturnOldManifestTempPath(t *testing.T) {
-	existing := getTestPodManifest(t).(*manifest)
-	updated := getUpdatedManifest(t).(*manifest)
+	existing := getTestPodManifest(t).GetBuilder()
+	updated := getUpdatedManifest(t).GetBuilder()
 
 	poddir, err := ioutil.TempDir("", "poddir")
 	Assert(t).IsNil(err, "couldn't create tempdir")
 	pod := NewPod(types.PodID("testPod"), poddir)
 
 	// set the RunAs user to the user running the test, because when we
-	// write files we need an owner. Unset them after the write so that the
-	// manifests match the manifests on disk
```

Comment:
I'm a little puzzled by the changes that were required here. It seems that the test as written before was expecting the `existing.Marshal()` call to not write the `RunAs:` user out. The test looks sane in its current form but i'm not sure why some of this was needed before and not now.